### PR TITLE
bpo-29677: DOC: clarify documentation for `round`

### PR DESCRIPTION
**Pull request by gerritholl**
Clarified that `round` can take a negative value for *ndigits*.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA. This is necessary for legal reasons before we can look at your contribution. Please follow these steps to help rectify the issue:

1. If you don't have an account on b.p.o, please [create one](http://bugs.python.org/user?@template=register)
2. Make sure your GitHub username is listed in ["Your Details" at b.p.o](https://cloud.githubusercontent.com/assets/2680980/23276970/d14a380c-f9d1-11e6-883d-e13b6b211239.png)
3. If you have not already done so, please sign the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/). The "bugs.python.org username " requested by the form is the "Login name" field under "Your Details".
4. If you just signed the CLA, please **wait at least one US business day** and then check "Your Details" on [bugs.python.org](https://bugs.python.org/) to see if your account has been marked as having signed the CLA (the delay is due to a person having to manually check your signed CLA)
5. Reply here saying you have completed the above steps


Thanks again to your contribution and we look forward to looking at it!

File: Doc/library/functions.rst [hunk 1]
```diff
@@ -1262,7 +1262,8 @@ are always available.  They are listed here in alphabetical order.
    closest multiple of 10 to the power minus *ndigits*; if two multiples are
```
Comment (vstinner) [inline Doc/library/functions.rst:1262]: while we are modifying the doc, I dislike the start: "Return the floating point value *number*": number can be an integer. round(123, -2) doesn't use floating point numbers. Maybe rephrase to following text?

Return  *number* rounded to (...). *number* can be an integer or a floating point number.

Or just remove "floating pointer number"?

**Comment by vstinner** [comment]
CC @JulienPalard

File: Doc/library/functions.rst (hunk 1 above)
Comment (JulienPalard) [reply Doc/library/functions.rst:1262]: @haypo: agree:

> Return *number* rounded to *ndigits* digits after the decimal point.

seems good to me.

About the negation, placement is good (too soon would have diluted the definition). About its wording, I'd prefer telling what it does instead of just stating it's possible, something like:

> Negative *ndigits* permit rouding before the decimal point.

Comment (vadmium) [reply Doc/library/functions.rst:1262]: Victor, I agree the first sentence is too specific. But the number is not restricted to “int” and “float”. I expect it should work with anything implementing `__round__`. This is already hinted by the rest of the entry, as well as

<https://docs.python.org/dev/reference/datamodel.html#object.__round__>
<https://docs.python.org/dev/library/numbers.html#numbers.Real>
<https://docs.python.org/dev/library/fractions.html#fractions.Fraction.__round__>

Comment (Mariatta) [reply Doc/library/functions.rst:1262]: Do you think we can accept this PR? If you would still like to see improvement with the first sentence, perhaps we can keep the issue open, so another person can work on it. Otherwise, @gerritholl's change in this PR seems good.

**Review by Mariatta — APPROVED** [review]

**Comment by Mariatta** [comment]
Thanks @gerritholl for your first contribution to CPython 🎉 
Thanks @haypo for merging this. I cherry-picked this into 3.5 and 3.6, but getting conflict in 2.7, so I won't bother with that branch 🤷‍♀️

File: Doc/library/functions.rst (hunk 1 above)
Comment (csabella) [reply Doc/library/functions.rst:1262]: I've made a few changes trying to clarify the wording for `round`.  I hope I didn't overstep any bounds.  I researched other, similar functions and looked at the help().  In the first line I wanted to say 'Return the real number *x*', but float() didn't specify real number, even though they both don't work on Complex.

Anyway, this is my first pull, so I hope I did OK.  :-)
